### PR TITLE
Steering Firmware Update and Steering Driver Creation

### PR DESCRIPTION
**Pull request by garethellis0**
- Updated the steering firmware with a bunch of debug options (print statements :laughing: )
- Created a driver for sending commands to the steering firmware

**NOTE TO SELF:** See cards on your TODO list before you merge this. Make sure you've covered everything

@ValRat - looking at this, I think we should be basically good to go. I don't want to merge it in to master without proper testing with a physical system though, so we should merge this into a `testing` branch off of master with any other relevant things for testing Thursday, and merge it after if all is good

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -89,7 +89,7 @@ if you're on campus use the `ubcsecure` or `resnet` networks for best results.
 - Variables are **non_camel_case**
 - Constants are **ALL_CAPS_WITH_UNDERSCORES**
 - Functions are **camelCase**
-- Indentations are 4 spaces
+- Indentations are 4 spaces, with the exception of arduino firmware files, which are 2 spaces
```
Comment (ValRat) [inline README.md]: Ah exception :X

File: src/drivers/src/SteeringDriver.cpp [hunk 2]
```diff
@@ -1,30 +1,85 @@
...
+            twist_msg->linear.z,
+    };
+    std::vector<double> angular = {
+            twist_msg->angular.x,
+            twist_msg->angular.y,
+            twist_msg->angular.z,
+    };
+
+    // Map the twist message values to ones the arduino can understand
+    for (double val : linear) map(val, -max_abs_linear_speed, max_abs_linear_speed, 0, 255);
```
Comment (ValRat) [inline src/drivers/src/SteeringDriver.cpp]: witchery

File: src/firmware/differential_drive_controller/differential_drive_controller.ino [hunk 3]
```diff
@@ -80,41 +123,60 @@ void setup() {
...
 
 /*
 * Calculates OFFSET for the joystick controllers. 
 */
 void set_offset() {
-  int linear_x_mid = 1325; //RX standard - radio signal midpoint
-  int angular_z_mid = 1325; //RY standard - radio signal midpoint
+  delay(100);
+  int linear_x_mid = pulseIn(2,HIGH); //RX standard - radio signal midpoint
+  int angular_z_mid = pulseIn(3,HIGH); //RY standard - radio signal midpoint
```
Comment (Dulluhan) [inline src/firmware/differential_drive_controller/differential_drive_controller.ino]: a }

File: src/firmware/differential_drive_controller/differential_drive_controller.ino [hunk 4]
```diff
@@ -80,41 +123,60 @@ void setup() {
...
+    Serial.print("Angular X: ");Serial.println(angular_x);
+    Serial.print("Angular Y: ");Serial.println(angular_y);
+    Serial.print("Angular Z: ");Serial.println(angular_z);
+    Serial.println();
+  #endif
 }
 
 /*
 * Calculates OFFSET for the joystick controllers. 
 */
```
Comment (Dulluhan) [inline src/firmware/differential_drive_controller/differential_drive_controller.ino]: Include in operating procedure that the reset button should be pressed at the beginning of runs with the following conditions: 
1. E-stop down
2. Remote on in neutral throttle on both joysticks

**Comment by ValRat** [comment]
Confirm that this was superceded by arduino_ros_bridge?

**Comment by garethellis0** [comment]
Yes, it was. Closing

**Pull request closed without merging**